### PR TITLE
Test all maps in parallel integration tests

### PR DESCRIPTION
**Pull request by Tastyfish**
## About The Pull Request

All maps are now run through parallel integration tests as a sanity check.
- To make doing so more useful, I've added a preamble unit test that redlines all of the `log_mapping()` calls that have station or mining AREACOORDS as non-coord'd errors are useful from a unit testing perspective and space coordinates are largely useless.
- This can probably replace those arcane map linters if all of them have useful map logs as well, I haven't confirmed yet.
- Note that this means that general unit test failures will be repeated 7 times (one for each map), but it's impossible to be sure if a bug is due to the map or generic code stuff.

This adds about 10-15 seconds to the total CI time as that's how long it takes to scrape the map names and bootstrap the CI matrix.

**Note:** I'm marking this PR as a draft because icebox and kilo station have legitimate failures that need to be fixed first.

### Preview, but also actual problems in master as foreshadowed above:
image

## Why It's Good For The Game

I actually found and fixed 3 runtimes in seperate PR's just testing this, let alone the 7 map issues shown in the image.

## Changelog

N/A

### CODE REVIEW

**Review by san7890 — COMMENTED** [review]
Thank you for your strong work, and I eagerly anticipate the implementation of this tool.

As far as the map errors you have mentioned in this PR, the merging of #66805 (which should be in the coming hours from this time I have posted this) should have resolved the issues in regards to Kilo (I reviewed it on my local machine and no errors showed up on the branch, so I don't foresee anything happening in the future).

Funnily enough, some of those errors were 100% caused by myself in #66222, so to say that this would be invaluable is an understatement.

I'll look into the duplicate APC in the prison wing, but I'll check in with the lead mapper for the access helpers project on the other issue that's cropping up with Ice Box to see if that's not already going to be fixed since I'm not sure off the top of my head.

Addendum: #66866 will fix the APC error with Ice Box.

**Comment by Maurukas** [comment]
Wow. I was not anticipating something like this PR showing up in my inbox today.  This will directly detect a large number of issues that inevitably creep back into maps. Some of these are difficult to notice when reviewing (as shown by my merging of the PR that added the current failure). 

Thank you for this. This permanently improves mapping on the repo.

**Comment by Tastyfish** [comment]
> Thank you for your strong work, and I eagerly anticipate the implementation of this tool.
> 
> As far as the map errors you have mentioned in this PR, the merging of #66805 (which should be in the coming hours from this time I have posted this) should have resolved the issues in regards to Kilo (I reviewed it on my local machine and no errors showed up on the branch, so I don't foresee anything happening in the future).
> 
> Funnily enough, some of those errors were 100% caused by myself in #66222, so to say that this would be invaluable is an understatement.
> 
> I'll look into the duplicate APC in the prison wing, but I'll check in with the lead mapper for the access helpers project on the other issue that's cropping up with Ice Box to see if that's not already going to be fixed since I'm not sure off the top of my head.
> 
> Addendum: #66866 will fix the APC error with Ice Box.

Awesome, I suspected that was the case since I did see #66805, but wasn't sure if those ones were addressed. Thanks for addressing the issues.

**Comment by Tastyfish** [comment]
@Mothblocks All maps now pass (this current standard) amazingly.

File: code/modules/unit_tests/mapping.dm [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/datum/unit_test/log_mapping
```
Comment (Mothblocks) [inline code/modules/unit_tests/mapping.dm:1]: Doc-comment explaining what this does

File: code/modules/unit_tests/unit_test.dm [hunk 2]
```diff
@@ -14,6 +14,7 @@ You can use the run_loc_floor_bottom_left and run_loc_floor_top_right to get tur
 GLOBAL_DATUM(current_test, /datum/unit_test)
 GLOBAL_VAR_INIT(failed_any_test, FALSE)
 GLOBAL_VAR(test_log)
+GLOBAL_LIST_EMPTY(unit_test_mapping_logs)
```
Comment (Mothblocks) [inline code/modules/unit_tests/unit_test.dm:17]: Doc-comment because this is a public API (used by mapping.dm)

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/modules/unit_tests/mapping.dm (hunk 1 above)
Comment (Tastyfish) [reply code/modules/unit_tests/mapping.dm:2]: done

File: code/modules/unit_tests/unit_test.dm (hunk 2 above)
Comment (Tastyfish) [reply code/modules/unit_tests/unit_test.dm:18]: done

**Pull request merged**